### PR TITLE
Csharp/feature/use generic math

### PR DESCRIPTION
**Pull request by FreePhoenix888**

### CODE REVIEW

File: csharp/Platform.Data.Tests/LinksConstantsTests.cs [hunk 1]
```diff
@@ -41,15 +42,15 @@ public static void ExternalReferencesTest()
...
-            var unsingedOne = Arithmetic.Increment(default(TUnsigned));
+            var unsingedOne = TUnsigned.One;
             var converter = UncheckedConverter<TSigned, TUnsigned>.Default;
             var half = converter.Convert(NumericType<TSigned>.MaxValue);
-            LinksConstants<TUnsigned> constants = new LinksConstants<TUnsigned>((unsingedOne, half), (Arithmetic.Add(half, unsingedOne), NumericType<TUnsigned>.MaxValue));
+            LinksConstants<TUnsigned> constants = new LinksConstants<TUnsigned>((unsingedOne, half), (half+ unsingedOne, NumericType<TUnsigned>.MaxValue));
 
             var minimum = new Hybrid<TUnsigned>(default, isExternal: true);
-            var maximum = new Hybrid<TUnsigned>(half, isExternal: true);
+            var maximum = new Hybrid<TUnsigned>((TUnsigned.One/(TUnsigned.One+TUnsigned.One)), isExternal: true);
```
Comment (Konard) [inline csharp/Platform.Data.Tests/LinksConstantsTests.cs:53]:
```suggestion
            var maximum = new Hybrid<TUnsigned>(half, isExternal: true);
```

**Review by Konard — CHANGES_REQUESTED** [review]

File: csharp/Platform.Data.Tests/LinksConstantsTests.cs [hunk 2]
```diff
@@ -41,15 +42,15 @@ public static void ExternalReferencesTest()
             TestExternalReferences<ushort, short>();
             TestExternalReferences<byte, sbyte>();
         }
-        private static void TestExternalReferences<TUnsigned, TSigned>()
+        private static void TestExternalReferences<TUnsigned, TSigned>() where TUnsigned : IUnsignedNumber<TUnsigned>
         {
-            var unsingedOne = Arithmetic.Increment(default(TUnsigned));
+            var unsingedOne = TUnsigned.One;
             var converter = UncheckedConverter<TSigned, TUnsigned>.Default;
             var half = converter.Convert(NumericType<TSigned>.MaxValue);
```
Comment (Konard) [inline csharp/Platform.Data.Tests/LinksConstantsTests.cs:49]: Is there a way to make generic conversions in .NET 7, converters may be slower than native conversion.

File: csharp/Platform.Data.Tests/LinksConstantsTests.cs (hunk 1 above)
Comment (FreePhoenix888) [reply csharp/Platform.Data.Tests/LinksConstantsTests.cs:53]: How to compute half?

**Review by Konard — CHANGES_REQUESTED** [review]

File: csharp/Platform.Data.Tests/LinksConstantsTests.cs (hunk 1 above)
Comment (Konard) [reply csharp/Platform.Data.Tests/LinksConstantsTests.cs:53]: It is already computed above.

File: csharp/Platform.Data.Tests/LinksConstantsTests.cs (hunk 2 above)
Comment (FreePhoenix888) [reply csharp/Platform.Data.Tests/LinksConstantsTests.cs:49]: What about 
```
NumericType<TUnsigned>.MaxValue / 2
```
?

Comment (Konard) [reply csharp/Platform.Data.Tests/LinksConstantsTests.cs:49]: @FreePhoenix888 division is slow, we already have required value in NumericType<TSigned>.MaxValue

**Pull request merged**